.circleci/config.yml

```diff
@@ -82,6 +82,7 @@ jobs:
     working_directory: ~/go/src/github.com/Azure/open-service-broker-azure
     environment:
       GOPATH: ~/go
+      REGISTRY: docker.io/microsoft/
```

Comment:
We can actually drop the leading `docker.io`, because that's the default registry for Docker anyway.

At that point, what we have is `REGISTRY: microsoft` and that looks funny, because the value doesn't look like a registry. It looks like a _repository_. This is a smell that's telling me we got something wrong here in the first place.

This is a pattern we borrowed from older projects. I don't want to blindly follow it if it doesn't make sense.

Through a conversation with @carolynvs, I think I've come to realize that we should rename this env var as `DOCKER_REPO`. That way an undefined value, `microsoft/` or `$(docker-machine ip my-docker):5000/` all equally make sense-- as even though some contain a host/ip name and some do not, they consistently all do point to _a repository_.

So... can I ask that we switch from `REGISTRY` to `DOCKER_REPO` and make all requisite changes to the `Makefile` as well?

The end result will be much cleaner and easier to understand than where we stand today.

Reply:
SGTM. I'll update the Makefile as well.

Reply:
I like it!